compiler/rustc_mir_build/src/thir/pattern/const_to_pat.rs

```diff
@@ -468,6 +479,10 @@ fn type_has_partial_eq_impl<'tcx>(
     let partial_eq_trait_id = tcx.require_lang_item(hir::LangItem::PartialEq, None);
     let structural_partial_eq_trait_id = tcx.require_lang_item(hir::LangItem::StructuralPeq, None);
 
+    if ty.has_escaping_bound_vars() {
+        return (false, false, false, None);
+    }
```

Comment:
is this necessary given that you bail in the `UsedParamsNeedInstantiationVisitor`?

I'd expect all other callers to not provide a type with escaping bound vars. MAybe just add `debug_assert!(!ty.has_escaping_bound_vars())` to the start of this function